pybsn/__init__.py

```diff
@@ -197,26 +335,47 @@ def __init__(self, url, session):
         self.session = session
         self.root = Node("controller", self)
 
-    def get(self, path, params=None):
+    def set_default_timeout(self, timeout: Optional[TimeoutSauce]):
+        """ Change the default timeout for operational responses.
+
+        Parameters:
+            :param timeout: New time out value.  None indicates to wait
+            until the response arrives, without timing out.
+        """
+        self.session.timeout = timeout
+
+    def get_default_timeout(self) -> Optional[TimeoutSauce]:
+        return self.session.timeout
+
+    def get(self, path, params=None,
+            timeout: TimeoutOverride = CLIENT_TIMEOUT):
         """ Retrieves information from the REST API using the GET method.
 
         :param path: the URL path to retrieve the data from; does not include the prefix
               (/api/v1/data).
         :param params: request parameters to attach
+        :param timeout: Amount of time to wait for response.  None indicates
+            to wait forever.  CLIENT_TIMEOUT indicates to use the default value
+            from BigDbClient
         :return: the deserialized result as list.
         """
-        return self._request("GET", path, params=params).json()
+        return self._request("GET", path, params=params, timeout=timeout).json()
 
-    def rpc(self, path, data, params=None):
+    def rpc(self, path, data, params=None, timeout: TimeoutOverride = CLIENT_TIMEOUT):
```

Comment:
While I like python type annotations as such, it seems broken / inconsistent to add them for the timeout when everything else does not have them. We also need to think about whether those work in all of the python versions that we claim support for.

Let's remove the type declarations and annotations in this PR, and have a separate Improvement Jira to track type-annotating the entire API (and then also consider whether that introduces any trade-offs in terms of python version / whether we need to bump the minimum supported version / bump the pybsn version to a new major release version etc.)